diff --git a/surf.1 b/surf.1
index 1715f93..c3a335d 100644
--- a/surf.1
+++ b/surf.1
@@ -6,6 +6,7 @@ surf \- simple webkit-based browser
 .RB [-bBdDfFgGiIkKmMnNpPsSvx]
 .RB [-a\ cookiepolicies]
 .RB [-c\ cookiefile]
+.RB [-C\ cachefolder]
 .RB [-e\ xid]
 .RB [-r\ scriptfile]
 .RB [-t\ stylefile]
@@ -40,6 +41,11 @@ Specify the
 .I cookiefile
 to use.
 .TP
+.B \-C cachefolder
+Specify the
+.I cachefolder
+to use.
+.TP
 .B \-d
 Disable the disk cache.
 .TP
diff --git a/surf.c b/surf.c
index 23c49bd..9236fdc 100644
--- a/surf.c
+++ b/surf.c
@@ -1641,7 +1641,7 @@ void
 usage(void)
 {
 	die("usage: %s [-bBdDfFgGiIkKmMnNpPsSvx] [-a cookiepolicies ] "
-	    "[-c cookiefile] [-e xid] [-r scriptfile] [-t stylefile] "
+	    "[-c cookiefile] [-C cachefolder] [-e xid] [-r scriptfile] [-t stylefile] "
 	    "[-u useragent] [-z zoomlevel] [uri]\n", basename(argv0));
 }
 
@@ -1691,6 +1691,9 @@ main(int argc, char *argv[])
 	case 'c':
 		cookiefile = EARGF(usage());
 		break;
+  case 'C':
+    cachefolder = EARGF(usage());
+    break;
 	case 'd':
 		enablediskcache = 0;
 		break;
